core-library/src/main/java/org/silverpeas/core/admin/component/model/ComponentInst.java

```diff
@@ -131,13 +138,29 @@ public static int getComponentLocalId(final String componentInstanceId) {
   }
 
   @Override
-  public Object clone() {
-    ComponentInst ci;
-    try {
-      ci = (ComponentInst) super.clone();
-    } catch (CloneNotSupportedException e) {
-      throw new SilverpeasRuntimeException(e);
-    }
+  public BasicIdentifier getIdentifier() {
+    return new BasicIdentifier(getLocalId(), getId());
+  }
+
+  public ComponentInst copy() {
```

Comment:
I agree with this. Nevertheless, the _ComponentInst_ is the bottom child of an already existing hierarchy and in this case, a copy-method is ok here. For some contexts, it is ok to have a copy-method whereas a copy-constructor should be used when such a copy functionality is provided by a base class. To be honest, the copy-method has been chosen to replace in a more direct way the previous clone method that is now strongly recommended to avoid.

However, to be more coherent, we should choose only one of the copy-way: either a copy-method or a copy-constructor.  This is why, I'm going to take into account your remark.